### PR TITLE
Fixes branch to Work as ServiceCatalog Broker

### PR DESCRIPTION
**Pull request by maddisondavid**
The target branch updates the code to make it compatible with ServiceCatalog however there were still some components missing.  This PR updates the branch with:
1. Java Code Fixes
2. Helm Chart updates to register as a Broker
3. Skaffold configuration to simplify development.

**serviceCatalog Flag**
The `values.yaml` in the chart includes a `serviceCatalog` setting to allow the chart to be installed  against CF or ServiceCatalog with just a single switch.

**Testing**
The branch has been tested on a Kubernetes cluster:
After installation:
```
$ svcat get brokers
               NAME                NAMESPACE                           URL                           STATUS
+--------------------------------+-----------+-----------------------------------------------------+--------+
  ecs-service-broker-client                    http://ecs-service-broker.default:9090                Ready
```
**Provision Bucket:**
```
$ svcat provision --class ecs-bucket --plan 5gb test-bucket
  Name:        test-bucket
  Namespace:   default
  Status:
  Class:       f3cbab6a-5172-4ff1-a5c7-72990f0ce2aa
  Plan:        8e777d49-0a78-4cf4-810a-b5f5173b019d

Parameters:
  No parameters defined



$ svcat get instances
     NAME       NAMESPACE                  CLASS                                   PLAN                   STATUS
+-------------+-----------+--------------------------------------+--------------------------------------+--------+
  test-bucket   default     f3cbab6a-5172-4ff1-a5c7-72990f0ce2aa   8e777d49-0a78-4cf4-810a-b5f5173b019d   Ready
```

**Bind Bucket:**
```
$ svcat bind test-bucket  
  Name:        test-bucket
  Namespace:   default
  Status:
  Secret:      test-bucket
  Instance:    test-bucket

Parameters:
  No parameters defined
$ svcat get bindings
     NAME       NAMESPACE    INSTANCE     STATUS
+-------------+-----------+-------------+--------+
  test-bucket   default     test-bucket   Ready
```

**Resulting Secret:**
```
$ kubectl get secret test-bucket -o=yaml
apiVersion: v1
kind: Secret
metadata:
  creationTimestamp: "2020-03-23T16:09:39Z"
  name: test-bucket
  namespace: default
type: Opaque
data:
  accessKey: bWFkZGlkLTZiNjdiYmE2LWI2MmEtNDA4YS1hMjMxLWJhMThlM2RkNWJhMQ==
  bucket: bWFkZGlkLTU3ZGVlNTJhLTk4YzItNBQyMC04ZWI0LThkZTg5NzJhYWQ5NQ==
  endpoint: aHR0cDovLzEwLjI0My44Ni43MDo5MDIw
  path-style-access: dHJ1ZQ==
  s3Url: aHR0cDovL21hZGRpZC02YjY3YmJhNi11NjJhLTQwOGEtYTIzMS1iYTE4ZTNkZDViYTE6NitJM1dtc2xSVGd6YW5NaGlWNnN3ZUdpNGdESUV0NkFPdlh2ajJyNEAxMC4yNDMuODYuNzA6OTAyMC9tYWRkaWQtNTdkZWU1MmEtOThjMi00ZDIwLThlYjQtOGRlODk3MmFhZDk1
  secretKey: NitJM1dtc2xSVGd6YW5NaG2WNnN3ZUdpNGdESUV0NkFPdlh2ajJyNA==
```

### CODE REVIEW

File: charts/templates/configmap.yaml [hunk 1]
```diff
@@ -1,13 +1,36 @@
...
 data:
   application.yml: |-
     ---
     broker:
-      object-endpoint: "https://object.ecstestdrive.com"
-      management-endpoint: "https://portal.ecstestdrive.com"
-      namespace: "131118670375936839"
-      replication-group: ecstestdrivegeo
-      prefix: kubetesting-
+      object-endpoint: "http://10.243.86.70:9020"
```
Comment (spiegela) [inline charts/templates/configmap.yaml]: We should probably exclude these testing values here before I merge ;-)  I'll replace these with Helm values parameters before I merge back to master.

File: src/main/java/com/emc/ecs/servicebroker/service/EcsService.java [hunk 2]
```diff
@@ -146,7 +146,9 @@ private boolean bucketExists(String id) throws EcsManagementClientException {
 
     UserSecretKey createUser(String id) {
         try {
+            logger.info(String.format("Creating user %s", prefix(id)));
```
Comment (spiegela) [inline src/main/java/com/emc/ecs/servicebroker/service/EcsService.java]: Can you set these to `debug`? before merge. I just don't want to forget about it

File: skaffold.yaml [hunk 3]
```diff
@@ -0,0 +1,16 @@
+apiVersion: skaffold/v2alpha3
```
Comment (spiegela) [inline skaffold.yaml]: If I'm not using skaffold on a cluster will this get picked up?  Should we add a param for this?

File: charts/templates/configmap.yaml (hunk 1 above)
Comment (maddisondavid) [reply charts/templates/configmap.yaml]: Values have now been converted to Helm values

File: skaffold.yaml (hunk 3 above)
Comment (maddisondavid) [reply skaffold.yaml:1]: `skaffold` is a client side tool (https://skaffold.dev/) that acts as a build and deployment tool by first building the Dockerfile and then deploying to the target cluster using the Helm charts.  It does not require anything `skaffold` specific thing to be placed on the cluster.

Let me know if you'd rather have the `skaffold` configuration removed.

File: src/main/java/com/emc/ecs/servicebroker/service/EcsService.java (hunk 2 above)
Comment (maddisondavid) [reply src/main/java/com/emc/ecs/servicebroker/service/EcsService.java:149]: done.

**Pull request merged**